interfaces/builtin/hugepages_control.go

```diff
@@ -0,0 +1,49 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const hugepagesControlSummary = `allows controlling hugepages`
+
+const hugepagesControlBaseDeclarationSlots = `
+  hugepages-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const hugepagesControlConnectedPlugAppArmor = `
```

Comment:
Is it desired that this snippet provides _write_ access without _read_ access to `/dev/` and `/run/`?

Reply:
When testing the interface, I found that with the write access, read was implied as well and it worked correctly.

Reply:
I have set read explicitly here, it is not clear to me by looking at apparmor docs if this is implicitly set by "wk" in some way.

Reply:
FYI, 'w' does *not* imply read, but it is possible that another interface already granted read.